drivers/android/transaction.rs

```diff
@@ -88,26 +87,20 @@ impl Transaction {
         let data_address = alloc.ptr;
         let file_list = alloc.take_file_list();
         alloc.keep_alive();
-        let mut tr = Pin::from(UniqueArc::try_new(Self {
-            // SAFETY: `spinlock_init` is called below.
-            inner: unsafe { SpinLock::new(TransactionInner { file_list }) },
+        let tr = UniqueArc::pin_init::<core::convert::Infallible>(pin_init!(Self {
+            inner: spinlock!(TransactionInner { file_list }, "Transaction::inner"),
             node_ref: None,
             stack_next: None,
             from: from.clone(),
             to,
             code: tr.code,
             flags: tr.flags,
-            data_size: tr.data_size as _,
+            data_size: tr.data_size as usize,
```

Comment:
Is this needed?

(To clarify, I'm not going into the question of whether it is better/worse to be explicit about the type, we can discuss that elsewhere. I'm asking if the new macros made things such that the compiler can't infer the types anymore.)

Reply:
Yes this is needed, otherwise we get a type inference error (have not investigated why).